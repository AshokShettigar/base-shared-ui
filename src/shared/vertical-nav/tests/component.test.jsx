import { mount } from 'enzyme';

import VerticalNav from '../component';

describe('Vertical Nav component', () => {
    const items = [{
        id: '1',
        title: 'Item 1',
        onClick: jest.fn()
    }, {
        id: '2',
        title: 'Item 2',
        onClick: jest.fn()
    }];
    const [activeItem] = items;
    const props = {
        activeItem,
        items
    };
    const wrapper = mount(<VerticalNav {...props} />);

    it('should render component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call "onClick" method', () => {
        jest.spyOn(activeItem, 'onClick');
        wrapper.find('.nav-link').first().simulate('click');
        expect(activeItem.onClick).toHaveBeenCalled();
    });
});
