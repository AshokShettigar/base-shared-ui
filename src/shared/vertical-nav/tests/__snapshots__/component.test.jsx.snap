// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Nav component should render component 1`] = `
<VerticalNav
  activeItem={
    Object {
      "id": "1",
      "onClick": [Function],
      "title": "Item 1",
    }
  }
  items={
    Array [
      Object {
        "id": "1",
        "onClick": [Function],
        "title": "Item 1",
      },
      Object {
        "id": "2",
        "onClick": [Function],
        "title": "Item 2",
      },
    ]
  }
>
  <Nav
    className="root"
    tag="ul"
    vertical={true}
  >
    <ul
      className="root nav flex-column"
    >
      <NavItem
        tag="li"
      >
        <li
          className="nav-item"
        >
          <NavLink
            className=""
            onClick={[Function]}
            tag="a"
          >
            <a
              className="nav-link"
              onClick={[Function]}
            >
              Item 1
            </a>
          </NavLink>
        </li>
      </NavItem>
      <NavItem
        tag="li"
      >
        <li
          className="nav-item"
        >
          <NavLink
            className=""
            onClick={[Function]}
            tag="a"
          >
            <a
              className="nav-link"
              onClick={[Function]}
            >
              Item 2
            </a>
          </NavLink>
        </li>
      </NavItem>
    </ul>
  </Nav>
</VerticalNav>
`;
